### PR TITLE
[CAS-491] Add link decorator

### PR DESCRIPTION
**Pull request by adasiewiczr**
Screenshot 2020-12-24 at 13 50 03

### Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Comparison screenshots added for visual changes
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt [hunk 1]
```diff
@@ -45,14 +46,25 @@ internal class BackgroundDecorator : BaseDecorator() {
             ShapeAppearanceModel.builder().setAllCornerSizes(radius).setBottomLeftCornerSize(bottomLeftCorner)
                 .setBottomRightCornerSize(bottomRightCorner).build()
         view.background = MaterialShapeDrawable(shapeAppearanceModel).apply {
+            val hasLink = data.message.attachments.any { it.hasLink() }
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/viewholder/decorator/BackgroundDecorator.kt:49]:
```suggestion
            val hasLink = data.message.attachments.any(::hasLink)
```

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_plain_text.xml [hunk 2]
```diff
@@ -8,6 +8,16 @@
     android:layout_marginEnd="8dp"
     >
 
+    <View
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_plain_text.xml:11]: Seems you don't need that. You can use messageContainer for the same purpose, can't you?

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemViewTypeMapper.kt [hunk 3]
```diff
@@ -30,14 +30,22 @@ internal object MessageListItemViewTypeMapper {
         return when {
             messageItem.message.deletedAt != null -> MessageListItemViewType.MESSAGE_DELETED
             messageItem.message.isGiphyEphemeral() -> MessageListItemViewType.GIPHY
-            messageItem.message.text.isNotEmpty() && messageItem.message.attachments.isMedia() -> MessageListItemViewType.PLAIN_TEXT_WITH_MEDIA_ATTACHMENTS
-            messageItem.message.text.isNotEmpty() && messageItem.message.attachments.isNotEmpty() -> MessageListItemViewType.PLAIN_TEXT_WITH_FILE_ATTACHMENTS
+            messageItem.message.text.isNotEmpty() && messageItem.message.attachments.isMediaOrLink() -> MessageListItemViewType.PLAIN_TEXT_WITH_MEDIA_ATTACHMENTS
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemViewTypeMapper.kt:33]: First of all, `isMediaOrLink` doesn't relate to the logic inside completely. If I understood it correctly I would call it `atLeastOneMedia`
Generally, I think all these conditions look hard to be understood. What if we made some func `Message.isMediaWithText()` and rewrite whole line as:
```kotlin
messageItem.message.isMediaWithText() -> MessageListItemViewType.PLAIN_TEXT_WITH_MEDIA_ATTACHMENTS
messageItem.message.isFileWithText() -> MessageListItemViewType.PLAIN_TEXT_WITH_FILE_ATTACHMENTS
```

What do you think?

**Review by ogkuzmin — COMMENTED** [review]
LGTM. Good job! left a few comments

File: stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_plain_text.xml (hunk 2 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-components/src/main/res/layout/stream_ui_item_message_plain_text.xml:11]: right, thanks!

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemViewTypeMapper.kt (hunk 3 above)
Comment (adasiewiczr) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/adapter/MessageListItemViewTypeMapper.kt:33]: Basically there are 3 cases:
1. Message has only media or link attachments but at least one of them is media - PLAIN_TEXT_WITH_MEDIA
2. Message has attachments and at least one of them is not link - PLAIN_TEXT_WITH_FILE
3. PLAIN_TEXT

I agree, it's quite complicated. I'll improve that

**Review by ogkuzmin — APPROVED** [review]
Nice 👍

**Pull request merged**